pkgs/development/libraries/doctest/default.nix

```
@@ -11,6 +11,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-hotO6QVpPn8unYTaQHFgi40A3oLUd++I3aTe293e4Aw=";
   };
 
+  patches = [
+    # Suppress unsafe buffer usage warnings with clang 16, which are treated as errors due to `-Werror`.
+    (fetchpatch {
+      url = "https://github.com/doctest/doctest/pull/768.patch";
```

```suggestion
      url = "https://github.com/doctest/doctest/commit/9336c9bd86e3fc2e0c36456cad8be3b4e8829a22.patch";
```

The pull request can change, so it's better to use a rev.

---

I pushed an update that uses the commit rather than the PR patch.